hopsworks-api/src/main/java/io/hops/hopsworks/api/pythonDeps/PythonDepsService.java

```diff
@@ -348,6 +464,28 @@ public Response removeEnv(@Context SecurityContext sc, @Context HttpServletReque
     return noCacheResponse.getNoCacheResponseBuilder(Response.Status.OK).build();
   }
 
+  @GET
+  @Path("/export")
+  @AllowedProjectRoles({AllowedProjectRoles.DATA_OWNER, AllowedProjectRoles.DATA_SCIENTIST})
+  public Response export(@Context SecurityContext sc,
+                           @Context HttpServletRequest req,
+                           @Context HttpHeaders httpHeaders) throws AppException {
+
+    String hdfsUser = getHdfsUser(sc);
+
+    String cpuHost = hostsFacade.findCPUHost();
+    if (cpuHost != null) {
+      exportEnvironment(cpuHost, "environment_cpu.yml", hdfsUser);
+    }
+
+    String gpuHost = hostsFacade.findGPUHost();
+    if (gpuHost != null) {
+      exportEnvironment(gpuHost, "environment_gpu.yml", hdfsUser);
+    }
+
```

Comment:
if both cpuHost and gpuHost are null, there will not be an error.
if (cpuHost ==null && gpuHost == null) {
   throw new AppExeception(.... "Export environment failed. Could not find any hosts.");
}